### PR TITLE
OpenSora v1.1 and Image/Video-to-Video inference

### PR DESCRIPTION
**Pull request by hadipash**

### CODE REVIEW

File: examples/opensora_cai/opensora/models/layers/blocks.py [hunk 1]
```diff
@@ -188,18 +261,24 @@ def __init__(
         dim,
         num_heads=8,
         qkv_bias=False,
+        qk_norm: bool = False,
         attn_drop=0.0,
         proj_drop=0.0,
+        norm_layer: nn.Cell = LlamaRMSNorm,
```
Comment (zhtmike) [inline examples/opensora_cai/opensora/models/layers/blocks.py:267]: Type[nn.Cell]

File: examples/opensora_cai/opensora/models/layers/blocks.py [hunk 2]
```diff
@@ -491,3 +621,122 @@ def construct(self, labels, train, force_drop_ids=None):
...
+        if train:
+            assert caption.shape[2:] == self.y_embedding.shape
+        use_dropout = self.uncond_prob > 0
+        if (train and use_dropout) or (force_drop_ids is not None):
+            caption = self.token_drop(caption, force_drop_ids)
+        caption = self.y_proj(caption)
+        return caption
+
+
+class PositionEmbedding2D(nn.Cell):
```
Comment (zhtmike) [inline examples/opensora_cai/opensora/models/layers/blocks.py:696]: Is this module support dynamic `dim` during training?

File: examples/opensora_cai/opensora/models/layers/rotary_embedding.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal  # FIXME: python 3.7
+
+import numpy as np
+
+from mindspore import Parameter, Tensor, dtype, nn, ops
+
+
+def rotate_half(x: Tensor) -> Tensor:
```
Comment (zhtmike) [inline examples/opensora_cai/opensora/models/layers/rotary_embedding.py:17]: existed in mindone/models/modules/pos_embed.py, maybe reuse

File: examples/opensora_cai/opensora/models/layers/blocks.py [hunk 4]
```diff
@@ -238,6 +317,12 @@ def construct(self, x, mask=None):
         k = ops.squeeze(k, axis=2)
         v = ops.squeeze(v, axis=2)
 
+        # WARNING: this may be a bug
```
Comment (zhtmike) [inline examples/opensora_cai/opensora/models/layers/blocks.py:320]: from my exp apply ROPE in fp32 gives a better convergence, you may have a try

File: examples/opensora_cai/sample_t2v.py [hunk 5]
```diff
@@ -266,9 +278,16 @@ def parse_args():
     parser.add_argument("--enable_dvm", default=False, type=str2bool, help="enable dvm mode")
     parser.add_argument("--sampling_steps", type=int, default=50, help="Diffusion Sampling Steps")
     parser.add_argument("--guidance_scale", type=float, default=8.5, help="the scale for classifier-free guidance")
+    parser.add_argument("--frame_interval", type=int)
```
Comment (zhtmike) [inline examples/opensora_cai/sample_t2v.py:281]: add description

File: examples/opensora_cai/opensora/models/layers/blocks.py [hunk 6]
```diff
@@ -2,13 +2,86 @@
...
+        """
+        LlamaRMSNorm is equivalent to T5LayerNorm
+        """
+        super().__init__()
+        # weight -> gamma: match the orig repo and fix the converter instead?
+        self.gamma = Parameter(np.ones(hidden_size).astype(np.float32))
+        self.variance_epsilon = eps
+
+    def construct(self, hidden_states: Tensor):
+        hidden_states = hidden_states.astype(ms.float32)
```
Comment (zhtmike) [inline examples/opensora_cai/opensora/models/layers/blocks.py:25]: not manually cast here, maybe add this in WhiteList

File: examples/opensora_cai/opensora/models/layers/blocks.py (hunk 2 above)
Comment (hadipash) [reply examples/opensora_cai/opensora/models/layers/blocks.py:696]: No, but I think the support can be added, if necessary.

File: examples/opensora_cai/opensora/models/layers/blocks.py (hunk 4 above)
Comment (hadipash) [reply examples/opensora_cai/opensora/models/layers/blocks.py:320]: Thanks, will pay attention to this when working on the training integration.

File: examples/opensora_cai/opensora/models/layers/blocks.py (hunk 6 above)
Comment (hadipash) [reply examples/opensora_cai/opensora/models/layers/blocks.py:25]: A good suggestion - will add `RMSNorm` to the whitelist.

File: examples/opensora_cai/opensora/models/layers/rotary_embedding.py (hunk 3 above)
Comment (hadipash) [reply examples/opensora_cai/opensora/models/layers/rotary_embedding.py:17]: The rotary embedding in `mindone/models/modules/pos_embed.py` is slightly different from the one used is OpenSora (it uses [this package](https://github.com/lucidrains/rotary-embedding-torch)). I think it's better to preserve the imported `RotaryEmbedding` class intact for now. Let me see later if I can merge them.

**Review by CaitinZhao — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

File: examples/opensora_hpcai/scripts/inference.py [hunk 7]
```diff
@@ -33,10 +32,16 @@
...
-def init_env(mode, device_target, enable_dvm=False):
+def init_env(mode, device_target, enable_dvm=False, debug: bool = False):
+    if debug and mode == ms.GRAPH_MODE:  # force PyNative mode when debugging
+        logger.warning("Debug mode is on, switching execution mode to PyNative.")
+        mode = ms.PYNATIVE_MODE
+
     ms.set_context(
         mode=mode,
         device_target=device_target,
+        gpu_config={"matmul_allow_tf32": True},
```
Comment (SamitHuang) [inline examples/opensora_hpcai/scripts/inference.py:43]: is it necessary?

File: examples/opensora_hpcai/scripts/utils.py
Comment (SamitHuang) [inline examples/opensora_hpcai/scripts/utils.py:1]: the utility code should be placed in `opensora/utils`

File: examples/opensora_hpcai/scripts/inference.py (hunk 7 above)
Comment (hadipash) [reply examples/opensora_hpcai/scripts/inference.py:43]: Not strictly necessary, just to align with [OpenSora repo](https://github.com/hpcaitech/Open-Sora/blob/c6cc021d612456455addc1b3164c1d43eeb33b97/scripts/inference-long.py#L149). Even without this setting results are visually good. Should I remove it or add a CLI argument for this?

File: examples/opensora_hpcai/scripts/utils.py
Comment (hadipash) [reply examples/opensora_hpcai/scripts/utils.py:1]: Many image and video preprocessing operations have been moved to `opensora/utils/cond_data.py`. Later on, they will be merged with training preprocessing and then removed.

**Review by SamitHuang — APPROVED** [review]

**Review by zhtmike — APPROVED** [review]

File: examples/opensora_hpcai/scripts/inference.py (hunk 7 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/scripts/inference.py:42]: better to remove for keeping stable

**Pull request merged**